### DISCUSSION TITLE
Support different naming convention for CSV files

### DISCUSSION THREAD

**Discussion by brockfanning** [open-sdg/open-sdg#1427] [Ideas]
**Is your feature request related to a problem? Please describe.**

Out of the box Open SDG can use CSV files as a data source, but requires that they be named "indicator_n-n-n" (where n-n-n is the indicator id). By contrast the metadata files are just "n-n-n". Requiring the "indicator_" for CSV files seems a bit arbitrary and inconsistent.

**Describe the solution that you need**

Let's make the naming convention more easily configurable, and then default to using "n-n-n" for both data and metadata.

**Why do you need this feature?**

For a more intuitive setup and maintenance experience.

**Comment by brockfanning** [comment]
@brockfanning To add more details on the pros/cons.

**Comment by brockfanning** [comment]
Pros:
1. Consistency: Right now the metadata files are named according to the indicator ID, but the data files are named with this "indicator_" in front. Standardizing around one or the other would be better for consistency. This potentially reduces confusion and simplifies documentation.
2. Fewer special-cases in code: This is at least [one special-case in our code](https://github.com/open-sdg/open-sdg/blob/1.4.0-dev/_includes/components/indicator/edit-buttons.html#L5-L12) which is required because of this existing naming convention. With a more consistent naming convention we could (eventually, at version 2.0.0) remove those special cases. 

Cons:
1. Because we have to maintain backwards compatibility, we can't just make the change and simplify our code. We have to actually *add* code to support a more flexible naming convention, and then *add* configuration for controlling the naming convention. So the overall complexity of the code will increase - at least until version 2.0.0, when we can finally remove deprecated stuff.
2. This takes some effort - and we could always just leave it as-is, with the plan of addressing it when/if it becomes a more urgent problem.

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning @AnnCorp one of the cons for this one, is that it would be very complex to do this whilst trying to maintain backwards compatibility. Is it something to consider implementing now for 2.0.0?

**Comment by LucyGwilliamAdmin** [comment]
Perhaps something to make optional during a version 2 release and could then consider making mandatory for 3.0.0
